### PR TITLE
Remove all Bad Sector Management related code from Smartfs

### PR DESCRIPTION
**Pull request by amogh-samsung**
- Current BSM logic is incomplete and flawed,
  causes redundancy in smart_writesector.
- BSM logic not put to use as of now, might
  be improved upon and re-implemented later.

Signed-off-by: Amogh Hassija <a.hassija@samsung.com>

### CODE REVIEW

File: build/configs/artik053/STDK/defconfig [hunk 1]
```diff
@@ -916,7 +916,6 @@ CONFIG_SMARTFS_ERASEDSTATE=0xff
 CONFIG_SMARTFS_MAXNAMLEN=32
 # CONFIG_SMARTFS_MULTI_ROOT_DIRS is not set
 CONFIG_SMARTFS_ALIGNED_ACCESS=y
-# CONFIG_SMARTFS_BAD_SECTOR is not set
```
Comment (sijupunnnoose) [inline build/configs/artik053/STDK/defconfig:919]: @Taejun-Kwon,  i understand BAD sector logic is not used, but should we remove this code completely?

File: os/fs/driver/mtd/smart.c [hunk 2]
```diff
@@ -4654,107 +4426,12 @@ static int smart_writesector(FAR struct smart_struct_s *dev, unsigned long arg)
...
-		}
-		/* Update releasecount for bad sector and freecount for the
-		 * newly allocated but bad physical sector. */
-
-		block = physsector / dev->sectorsPerBlk;
-#ifdef CONFIG_MTD_SMART_PACK_COUNTS
-		smart_add_count(dev, dev->releasecount, block, 1);
-		smart_add_count(dev, dev->freecount, physsector / dev->sectorsPerBlk, -1);
-#else
-		dev->releasecount[block]++;
```
Comment (thapav) [inline os/fs/driver/mtd/smart.c:4713]: same for release counts and other such variables ?

File: build/configs/artik053/STDK/defconfig (hunk 1 above)
Comment (Taejun-Kwon) [reply build/configs/artik053/STDK/defconfig:919]: we will not use it stand alone anymore, will be added inside of crc feature later.

File: os/fs/driver/mtd/smart.c [hunk 3]
```diff
@@ -4654,107 +4426,12 @@ static int smart_writesector(FAR struct smart_struct_s *dev, unsigned long arg)
...
-
-errout:
-#ifdef CONFIG_SMARTFS_BAD_SECTOR
-
-	if (bad_sector_found) {
-		needsrelocate = TRUE;
-		dev->badSectorList[physsector] = TRUE;
-
-#if CONFIG_SMARTFS_ERASEDSTATE == 0xFF
-		byte = header->status & ~(SMART_STATUS_RELEASED | SMART_STATUS_COMMITTED);
```
[the remark this answers is not in the document]
Comment (Taejun-Kwon) [reply os/fs/driver/mtd/smart.c:4694]: Basic logic is ...
write -> verify -> if it is not matched -> release sector -> add to bad sector list

File: os/fs/driver/mtd/smart.c (hunk 2 above)
Comment (Taejun-Kwon) [reply os/fs/driver/mtd/smart.c:4713]: As I wrote above, it released so count should be added.

**Pull request merged**